### ISSUE TITLE
Stringified objects are utf-8 encoded

### ISSUE THREAD

**Issue by jixam** [rurban/Cpanel-JSON-XS#210]
I expected identical handling of objects whether they use `TO_JSON` or a stringification overload but in fact the latter value gets utf-8 encoded (twice when `utf8` is set).

```
use Cpanel::JSON::XS;

use v5.36;

{
    package S;

    use overload '""' => sub { shift->{string}; };

    sub new($class, $string) { bless { string => $string } => $class };
}

{
    package J;

    our @ISA = 'S';

    sub TO_JSON { shift->{string}; }
}

my $unicode = "\x{1f603}";
my $JSON = Cpanel::JSON::XS->new->convert_blessed;

say length $JSON->utf8(0)->encode( [ $unicode ] );
say length $JSON->utf8(0)->encode( [ J->new($unicode) ] );
say length $JSON->utf8(0)->encode( [ S->new($unicode) ] );
say "";
say length $JSON->utf8(1)->encode( [ $unicode ] );
say length $JSON->utf8(1)->encode( [ J->new($unicode) ] );
say length $JSON->utf8(1)->encode( [ S->new($unicode) ] );
```

```
$ perl encode.pl
5
5
8

8
8
12
```

**Comment by rurban** [comment]
Well, if you encode it twice (TO_JSON plus string overload). I'll check if the overloading callback can be omitted for the TO_JSON case, but stacking both is the really the users fault. And I don't know if skipping the overload callback is ok then.

**Comment by jixam** [comment]
The problem is with `S` which is not stacked. The `TO_JSON` works as expected, even with an overload present.

Here is a (currently failing) testcase that I managed to create:
```
--- t/54_stringify.t
+++ t/54_stringify.t
@@ -34,6 +34,12 @@ my $enc = $json->encode( { obj => $object } );
 is( $enc, '{"obj":"Foo <foo>"}', "mg object stringified" )
   or diag($enc);

+$object = bless ["\x{1f603}"], 'Foo';
+$enc = $json->encode( { obj => $object } );
+
+is( $enc, '{"obj":"Foo <\x{1f603}>"}', "mg object stringified Unicode" )
+  or diag($enc);
+
 $enc = $json->encode( { time => $time } );
 isa_ok($time, "Time::Piece");
```

**Comment by rurban** [comment]
Ok, I'll check if I get my head around it

**Comment by rurban** [comment]
Fix coming today. Just need to fix 5.8 also

**Comment by jixam** [comment]
Thank you for the quick turnaround!
